### DISCUSSION TITLE
Add FILE_ATTRIBUTE_TEMPORARY to FileOptions

### DISCUSSION THREAD

**Discussion by msedi** [dotnet/runtime#65067] [Q&A]
From the WinAPI (https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-createfilea) I can see that there is a `FILE_ATTRIBUTE_TEMPORARY` flag when creating a file which seems to be very useful in my case.

The question in general is, does  `FILE_ATTRIBUTE_TEMPORARY` have an effect? I assume some other flags its only a hint to the file manager. I can also understand that this is not available on other platforms, but if it can be seen as a hint only (windows), it could also be seen as such in other platforms.

Another solution might be 

```cs
try
{                    
  File.SetAttributes(fileName, File.GetAttributes(fileName) | FileAttributes.Temporary);
}
catch
{
}
```

but this looks a bit clumsy.

**Comment by stephentoub** [comment]
> The problem is that obviously in FileOptions I cannot do something like this anymore as this was possible in the FileStream before the .NET 6 changes

Can you clarify? What code was possible pre-.NET 6 that breaks with .NET 6?

**Reply by msedi** [reply]
@stephentoub. You are right. I haven't tried this for a long time. It might have never worked. I'm not sure how I did this before, maybe I used CreateFile instead. Yesterday I saw in a blog in stackoverflow that people are suggesting it (e.g. for FILE_FLAG_NOBUFFERING) so I thought it would work.  So I updated the question to if it makes sense to enable the .NET api to support it.
